### PR TITLE
fix(column): modifying a sign should update placed signs

### PR DESCRIPTION
**Pull request by luukvbaal**
Problem:  Modifying a sign no longer updates already placed signs.
Solution: Loop over (newly-exposed) placed decorations when modifying a
          sign definition. Update placed decor if it belongs to the sign
          that is modified.

### CODE REVIEW

File: src/nvim/sign.c [hunk 1]
```diff
@@ -441,6 +435,27 @@ static int sign_define_by_name(char *name, char *icon, char *text, char *linehl,
     }
   }
 
+  // Update already placed signs and redraw if necessary when modifying a sign.
+  if (!new_sign) {
+    bool should_redraw = true;
+    for (size_t i = 0; i < kv_size(decor_items); i++) {
+      if (strequal(name, kv_A(decor_items, i).sign_name)) {
```
Comment (zeertzjq) [inline src/nvim/sign.c:442]: Check for `kSHIsSign` first?

Comment (zeertzjq) [reply src/nvim/sign.c:442]: Also, `kv_A(decor_items, i)` is used a lot of times here, so it may be worth assigning it to a variable.

Comment (luukvbaal) [reply src/nvim/sign.c:442]: > Check for `kSHIsSign` first?

`strequal` checks if `sign_name` is non-NULL first, which is a smaller set (i.e. legacy signs only) for which `kSHIsSign` should always be set.

**Comment by zeertzjq** [comment]
I've thought about this approach before, but didn't choose it, because `decor_items` may contain a lot of unrelated items (e.g. a lot of extmark signs).

**Comment by luukvbaal** [comment]
> I've thought about this approach before, but didn't choose it, because `decor_items` may contain a lot of unrelated items (e.g. a lot of extmark signs).

Yes, but it seems preferable to me as this only affects redefining a sign, which should be infrequent.

File: src/nvim/decoration.c [hunk 2]
```diff
@@ -293,6 +290,7 @@ static void decor_free_inner(DecorVirtText *vt, uint32_t first_idx)
     DecorSignHighlight *sh = &kv_A(decor_items, idx);
     if (sh->flags & kSHIsSign) {
       xfree(sh->sign_name);
+      sh->sign_name = NULL;  // Still referenced in "decor_items", avoid accessing.
```
Comment (zeertzjq) [inline src/nvim/decoration.c:293]: `XFREE_CLEAR`

**Pull request merged**